### DISCUSSION TITLE
Include en-dash and em-dash in Devilution X font?

### DISCUSSION THREAD

**Discussion by tsunamistate** [diasurgical/DevilutionX#4203] [Ideas]
En-dash (Alt+0150, –) and em-dash (Alt+0151, —) are longer variants of standard hyphen (Minus key, -).
Here's the article about when and where they should be used: https://www.grammarly.com/blog/hyphens-and-dashes/

As of now, these characters are rendered as spaces.
Would it be feasible to add these symbols to DevilutionX font?

**Comment by AJenbo** [comment]
Can you find the Unicode values for them instead of the Windows short-cuts?

**Reply by tsunamistate** [reply]
Sure, en-dash is U+2013, em-dash is U+2014

**Comment by AJenbo** [comment]
Page 20 is part of the extended font. It contains General Punctuation (of which en-dash and em-dash), Superscripts and Subscripts, Currency Symbols, and Combining Diacritical Marks for Symbols.

![image](https://user-images.githubusercontent.com/204594/159190007-c184f174-150a-4197-91bf-10aed614aad9.png)

https://github.com/diasurgical/devilutionx-assets/releases/download/v1/fonts.mpq
https://github.com/diasurgical/devilutionx-assets/blob/main/assets/fonts/46-20.pcx
https://en.wikibooks.org/wiki/Unicode/Character_reference/2000-2FFF

(note we currently do not have a kerning definition for page 20 which would be required for proper display)

We could look in to promoting this be be a part of the core assets if there is a good enough reason to do so. It would increase the size by 362KiB

**Reply by tsunamistate** [reply]
IMO, it would be beneficial to move at least characters from "General Punctuation" section (U+2000 – U+206F) from fonts.mpq to devilutionx.mpq

Characters that will be good to have in core set, at least to me:
* EN DASH and EM DASH as noted above
* HORIZONTAL ELLIPSIS instead of three periods
* DOUBLE LOW-9 QUOTATION MARK and DOUBLE HIGH REVERSED-9 QUOTATION MARK, as these quotation marks are the ones used in Ukrainian typography
* QUESTION EXCLAMATION MARK

**Reply by AJenbo** [reply]
We cannot move a partial page (256 symbols), that is the smallest unit we support. U2000 - U20FF

**Reply by tsunamistate** [reply]
That's fine by me :)

**Reply by AJenbo** [reply]
Alright, I will look in to it. I'm a big fan of HORIZONTAL ELLIPSIS my self :)

> DOUBLE LOW-9 QUOTATION MARK and DOUBLE HIGH REVERSED-9 QUOTATION MARK, as these quotation marks are the ones used in Ukrainian typography

That's a good argument since we have the rest of the relevant symbols in core.

**Reply by AJenbo** [reply]
The additional symbols are now part of the core: https://github.com/diasurgical/devilutionX/commit/69dbe0b3b2dd2e0583c655c99aedea53b9ab9d0f
